dicepy/__init__.py

```diff
@@ -15,6 +15,35 @@ class MathType(Enum):
     add = '+'
     subtract = '-'
 
+class Roll:
+    keep_type = None
+    keep_amount = 1
+    dice = None
+    value = None
+
+    def __init__(self, sides: int, dice_count: int, keep_type: KeepType = None, keep_amount: int = 1):
+        self.dice = [randint(1, sides) for _ in range(dice_count)]
+        self.keep_type = keep_type
+        self.keep_amount = keep_amount
+        if keep_type == None:
+            self.value = sum(self.dice)
+        # 3d6k2 and 3d6kh2
+        elif keep_type == KeepType.highest or keep_type == KeepType.keep:
+            self.value = sum(sorted(self.dice)[-self.keep_amount:])
+        else: # KeepType.lowest
+            self.value = sum(sorted(self.dice)[:self.keep_amount])
+
+        # Just for fun if this ever gets min'd to 3.10
+        #match keep_type:
+        #    case None: # KeepType.all?
+        #        self.value = sum(self.dice)
+        #    case KeepType.keep | KeepType.highest:
+        #        self.value = sum(sorted(self.dice)[-self.keep_amount:])
+        #    case KeepType.lowest:
+        #        self.value=sum(sorted(self.dice)[:self.keep_amount])
+
+    def __repr__(self):
+        return f"Roll(keep_type='{self.keep_type}' dice='{self.dice}' value='{self.value}')"
```

Comment:
Maybe do an extra piece here:

```python
amount = ' '
if self.keep_type:
    amount = f' keep_amount={self.keep_amount} '
return f'Roll(keep_type="{self.keep_type}"{amount}dice="{self.dice}" value="{self.value}")'
```
This lets you know how many are kept in the repr